test/e2e/util/mongotester/mongotester.go

```diff
@@ -0,0 +1,237 @@
+package mongotester
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+	"testing"
+	"time"
+
+	mdbv1 "github.com/mongodb/mongodb-kubernetes-operator/pkg/apis/mongodb/v1"
+	f "github.com/operator-framework/operator-sdk/pkg/test"
+	"github.com/stretchr/testify/assert"
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/options"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/wait"
+)
+
+type Tester struct {
+	mongoClient *mongo.Client
+	clientOpts  []*options.ClientOptions
+}
+
+func NewTester(opts ...*options.ClientOptions) (*Tester, error) {
+	t := &Tester{}
+	t.clientOpts = append(t.clientOpts, opts...)
+	return t, nil
+}
+
+// FromResource returns a Tester instance from a MongoDB resource. It infers SCRAM username/password
+// and the hosts from the resource.
+// NOTE: Tls is not configured as the mechanism that the ClientOptions are merged only merge on non-nil
+// values, meaning we need to remove option that configures TLS from the list if we want to not use tls.
+// For now we can just explicitly pass WithTls() or WithoutTls() to configure TLS.
+func FromResource(t *testing.T, mdb mdbv1.MongoDB, opts ...*options.ClientOptions) (*Tester, error) {
+	var clientOpts []*options.ClientOptions
+	clientOpts = append(clientOpts, WithHosts(mdb.Hosts()))
+	t.Logf("Configuring hosts: %s for MongoDB: %s", mdb.Hosts(), mdb.NamespacedName())
+
+	users := mdb.Spec.Users
+	if len(users) == 1 {
```

Comment:
[q] it's not clear to me why we make this check. Is it fundamental to have only one user? What happens if we have multiple?

Reply:
The idea here was that if we have a test which configures multiple users, we would need to be more explicit with which user we want to attempt to connect as (via `WithScram(username,password)`)

Since all of our tests right now just have a single user, it's just a short hand for that! If we start adding tests with many users, maybe we can make some improvements here